### PR TITLE
Feat: update ray requirement to 2.24.0

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
We were far behind in Ray versions

# This PR's solution
Bump ray to newest version

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: projects/orquestra-sdk/docs/examples/tests/test_parametrized_workflows.py [hunk 1]
```diff
@@ -126,7 +126,9 @@ def test_get_results(change_test_dir, shared_ray_conn, change_db_location):
         # When
         # Call the snippet
         proc = subprocess.run([sys.executable, str(src_file)], capture_output=True)
-
+        assert (
+            False
+        ), f"STDOUT: {proc.stdout.decode()}, \n STDERR: {proc.stderr.decode()}"
```
Comment (BenjaminMummery) [inline projects/orquestra-sdk/docs/examples/tests/test_parametrized_workflows.py:131]: Query: would you mind explaining what this is for?

**Review by BenjaminMummery — COMMENTED** [review]
LGTM, just one bit that I couldn't immediately make sense of

File: projects/orquestra-sdk/docs/examples/tests/test_parametrized_workflows.py (hunk 1 above)
Comment (SebastianMorawiec) [reply projects/orquestra-sdk/docs/examples/tests/test_parametrized_workflows.py]: Yeh sorry, python tests on 3.9 do not pass for some reason on CI. Im RN debugging why. Should've taken this PR to draft for that

File: projects/orquestra-sdk/docs/examples/tests/test_parametrized_workflows.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ################################################################################
```
Comment (BenjaminMummery) [inline projects/orquestra-sdk/docs/examples/tests/test_parametrized_workflows.py:3]: Nitpick: the above copyright string needs updating

**Review by BenjaminMummery — APPROVED** [review]
LGTM, one nitpick only :)

**Pull request merged**